### DISCUSSION TITLE
Suggestion: Self-Signed Certificate without domain

### DISCUSSION THREAD

**Discussion by Maelep** [vilhelmprytz/pterodactyl-installer#154] [Ideas]
Hello,

I discovered your insane script and would like to thank you for this one! Thank you!
I'm using pterodactyl but I don't use domain so, to get a certificate, I have to use openssl.
Could you integrate this possibility in your script?
Here is the tutorial I use to create it: [DigitalOcean: How to create a self signed SSL Certificate for NGINX in Ubuntu 16.04](https://www.digitalocean.com/community/tutorials/how-to-create-a-self-signed-ssl-certificate-for-nginx-in-ubuntu-16-04)

Thank you a lot!

**Comment by vilhelmprytz** [comment]
Hello! May I ask in what way a self-signed certificate is useful? I mean, no browser will accept it as secure unless the user installs your CA :slightly_smiling_face: The script currently allows you to "assume SSL" which means it will place an SSL "ready" Nginx configuration on your server. See this: https://github.com/vilhelmprytz/pterodactyl-installer/blob/600d8dc78adb0b0069618a5b03cd7469e6778c2c/configs/nginx_ssl.conf#L26-L27

If you opt in to "assume SSL", but not automatic Let's Encrypt, you could generate your self-signed certificates and update the config file accordingly.
